nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -109,9 +111,23 @@ def _setup(self, config):
     def _train(self):
         self.logger.debug(f"_train: {self._trial_info.trial_name}({self.iteration})")
         try:
-            # Check if restore checkpoint file fulfills the stop criteria on first run
             if self._first_run:
                 self._first_run = False
+
+                if self._validate_immediately:
```

Comment:
This is not necessary before the first epoch as it was originally intended. 
During the restore the `first_run` is the last epoch stored with the checkpoint